rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/DatabaseSanitation/DatabaseSanitationController.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright Siemens AG, 2023-2024.
+ * Part of the SW360 Portal Project.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+package org.eclipse.sw360.rest.resourceserver.DatabaseSanitation;
+
+import static org.springframework.hateoas.server.mvc.WebMvcLinkBuilder.linkTo;
+
+import java.util.List;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.apache.poi.ss.formula.functions.T;
+import org.apache.thrift.TException;
+import org.eclipse.sw360.datahandler.thrift.components.Release;
+import org.eclipse.sw360.datahandler.thrift.projects.Project;
+import org.eclipse.sw360.datahandler.thrift.users.User;
+import org.eclipse.sw360.rest.resourceserver.core.RestControllerHelper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.rest.webmvc.BasePathAwareController;
+import org.springframework.data.rest.webmvc.RepositoryLinksResource;
+import org.springframework.hateoas.CollectionModel;
+import org.springframework.hateoas.EntityModel;
+import org.springframework.hateoas.server.RepresentationModelProcessor;
+import org.springframework.http.HttpStatus.Series;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+
+@BasePathAwareController
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class DatabaseSanitationController  implements RepresentationModelProcessor<RepositoryLinksResource> {
+    
+    public static final String DATABASESANITATION_URL = "/dataSanitation";
+    
+    @NonNull
+    private final RestControllerHelper restControllerHelper;
+    
+    @NonNull
+    Sw360DatabaseSanitationService dataSanitationService;
+
+    @Override
+    public RepositoryLinksResource process(RepositoryLinksResource resource) {
+        resource.add(linkTo(DatabaseSanitationController.class).slash("api" + DATABASESANITATION_URL).withRel("dataSanitation"));
+        return resource;
+    }
+    @RequestMapping(value = DATABASESANITATION_URL + "/searchDublicate", method = RequestMethod.GET)
+    public ResponseEntity<CollectionModel<?>> searchDublicateIdentifiers(HttpServletRequest request )throws TException {
+        try {
+            User sw360User = restControllerHelper.getSw360UserFromAuthentication();
+            dataSanitationService.dublicateIdentifiers(sw360User, request);
+        } catch (Exception e) {
+            throw new TException(e.getMessage());
+        }
+        HttpStatus status = request == null ? HttpStatus.NO_CONTENT : HttpStatus.OK;
```

Comment:
Request is always non null here. This seems to be not valid check

Reply:
done.